app/renderer/components/Inspector/Inspector.js

```diff
@@ -147,103 +107,110 @@ export default class Inspector extends Component {
-      this.mjpegStreamCheckInterval = null;
-    }
-  }
-
-  screenshotInteractionChange (mode) {
-    const {selectScreenshotInteractionMode, clearSwipeAction} = this.props;
+  const screenshotInteractionChange = (mode) => {
+    const { selectScreenshotInteractionMode, clearSwipeAction } = props;
     clearSwipeAction(); // When the action changes, reset the swipe action
     selectScreenshotInteractionMode(mode);
-  }
+  };
 
-  render () {
-    const {screenshot, screenshotError, selectedElement = {},
-           quitSession, showRecord,
-           screenshotInteractionMode, visibleCommandMethod,
-           selectedInteractionMode, selectInteractionMode, setVisibleCommandResult,
-           showKeepAlivePrompt, keepSessionAlive, sourceXML, t, visibleCommandResult,
-           mjpegScreenshotUrl, isAwaitingMjpegStream, toggleShowCentroids, showCentroids,
-           isGestureEditorVisible, toggleShowAttributes, isSourceRefreshOn
-    } = this.props;
-    const {path} = selectedElement;
+  useEffect(() => {
+    const { applyClientMethod, getSavedActionFramework, runKeepAliveLoop, setSessionTime } = props;
+    const curHeight = window.innerHeight;
+    const curWidth = window.innerWidth;
+    const needsResize = (curHeight < MIN_HEIGHT) || (curWidth < MIN_WIDTH);
+    if (!didInitialResize.current && needsResize) {
+      const newWidth = curWidth < MIN_WIDTH ? MIN_WIDTH : curWidth;
+      const newHeight = curHeight < MIN_HEIGHT ? MIN_HEIGHT : curHeight;
+      // resize width to something sensible for using the inspector on first run
+      window.resizeTo(newWidth, newHeight);
+    }
+    didInitialResize.current = true;
+    applyClientMethod({methodName: 'getPageSource', ignoreResult: true});
+    getSavedActionFramework();
+    runKeepAliveLoop();
+    setSessionTime(Date.now());
+  }, []);
 
-    const showScreenshot = ((screenshot && !screenshotError) ||
-                            (mjpegScreenshotUrl && (!isSourceRefreshOn || !isAwaitingMjpegStream)));
+  /**
+   * Ensures component dimensions are adjusted only once windowSize exists.
+   * Cannot be combined with the other useEffect hook, since inside it,
+   * windowSize is set to 'undefined', and the event listener and MJPEG checker
+   * would not update this value when invoked
+   */
+  useEffect(() => {
+    if (windowSize) {
+      updateSourceTreeWidthDebounced();
+      window.addEventListener('resize', updateSourceTreeWidthDebounced);
+      if (mjpegScreenshotUrl) {
+        mjpegStreamCheckInterval.current = setInterval(checkMjpegStream, MJPEG_STREAM_CHECK_INTERVAL);
+      }
+    }
+    return () => {
+      if (windowSize) {
+        window.removeEventListener('resize', updateSourceTreeWidthDebounced);
+        if (mjpegStreamCheckInterval.current) {
+          clearInterval(mjpegStreamCheckInterval.current);
+          mjpegStreamCheckInterval.current = null;
+        }
+      }
+    };
+  }, [windowSize && windowSize.width]);
```

Comment:
I'm a little confused by this. Isn't this where we're supposed to pass the props that this effect depends on? But `windowSize && windowSize.width` is a boolean, not a prop?

I guess I can see how it works, but I wonder if the code might be a little less arcane if the dependency was just `windowSize` and the boolean check was put into the conditional inside the effect instead.

It's up to you, this just caught my eye.

Reply:
I did originally use just `windowSize`, but the issue with that was that the hook was fired every time `windowSize` was updated (even if the property values did not change), because it is an object and not a primitive type. Ideally this dependency would be just on `windowSize.width`, but the initial state of `windowSize` is undefined. So, I researched a better approach and found this, which also works.
However, I do agree that it's not very obvious. I searched again, and another approach for relatively simple objects (like in this case) is to just wrap the object in `JSON.stringify`. It does appears to work in my tests, so I will change this.

Reply:
ok, that all makes sense!